### PR TITLE
Add initial support for running tests on Open Data Hub

### PR DESCRIPTION
**Pull request by jgarciao**
Note: this PR needs to be merged at the same time as ods/odhcluster/-/merge_requests/49

Changes:
- Adds tag OpenDataHub to tag compatible tests (used in ODS-316 as example)
- Adds new global variables: PRODUCT, APPLICATIONS_NAMESPACE, MONITORING_NAMESPACE, OPERATOR_NAMESPACE, NOTEBOOKS_NAMESPACE
- Fails RHOSi Setup if the previous variables are not defined
- Adds variable file ods_ci/test-variables-odh-overwrite.yml to overwrite values in variables defined in tests suites to make them compatible with ODH

### CODE REVIEW

File: ods_ci/tests/Resources/OCP.resource [hunk 1]
```diff
@@ -64,7 +64,8 @@ Verify Notebook CR Is Running
 Is RHODS Self-Managed
     [Documentation]     Returns ${FALSE} if RHODS Managed (i.e., Cloud version) is installed
     ...                 in the cluster (e.g., usually on OSD, ROSA).
-    ...                 Returns ${TRUE} if RHODS Self-Managed is installed (e.g., usually on PSI or baremetal)
+    ...                 Returns ${TRUE} if RHODS Self-Managed is installed or if PRODUCT=ODH
+    IF  "${PRODUCT}" == "ODH"  RETURN     ${TRUE}
```
Comment (lugi0) [inline ods_ci/tests/Resources/OCP.resource:68]: For the final return value, we could just do
```
Run Keyword And Return    Evaluate    not ${is_managed}
```
see https://robotframework.org/code/?codeProject=N4IgdghgtgpiBcIDCBXAzgFwPZQAQGMsATOAGhBLXwCcBLABw1qzARBHIDNaAbGNBAG1Q3PgDlocRBn4YAdNSwAjLBg4hCYGVrYAqfbgBqEOhCV80ufboA6YAGTAAYgHkXAX1xfcAaxgBPAEYAXgASYAAVahQYT28-fwAmMOcIHjRYuwABYAAPfwAvOK8EwO9fAMTyhIBmOztrXAjZXCQIDMtrO2bMXEC7ctcAJXLw3OLcAEkxcpz8ooHy70mnUdzcWjBcUNcXXAgwIm3duTH3RaWlgBksAHMmrFaWNCw+cqh-AH1CKCgDo4uSwAolcAMpAwGXLw3e4RR5IZ6vGDvL6qAAWMGo3xwf0OkO8QLEABFAYSiSB3ABdcgkOgANxgRAACooAFYwfBqeCcNIZciKFQYQyYtDMViIACscgADHJAhSgA

**Review by lugi0 — APPROVED** [review]

File: ods_ci/tests/Resources/OCP.resource (hunk 1 above)
Comment (jgarciao) [reply ods_ci/tests/Resources/OCP.resource:68]: I've added it. Thanks for the suggestion!

**Comment by jgarciao** [comment]
Tested successfully here (on ODH): rhods-ci-pr-test/1970

Being tested right not (on RHODS): rhods-ci-pr-test/1971

**Review by lugi0 — DISMISSED** [review]

**Comment by jgarciao** [comment]
Tested successfully here:
- on ODH: rhods-ci-pr-test/1976  (test with OpenDataHub tag PASS)
- on RHODS: rhods-ci-pr-test/1977 (all Smoke PASS)

**Review by diegolovison — APPROVED** [review]

File: ods_ci/tests/Resources/RHOSi.resource [hunk 2]
```diff
@@ -71,3 +72,11 @@ Initialize Global Variables
     [Arguments]    ${force_fetch}=False
     ${RHODS_VERSION}=    Get RHODS Version    ${force_fetch}
     Set Global Variable   ${RHODS_VERSION}
+
+Required Global Variables Should Exist
+    [Documentation]   Fails if new required global variables are not set
+    Variable Should Exist    ${PRODUCT}
```
Comment (bdattoma) [inline ods_ci/tests/Resources/RHOSi.resource:78]: what if it's empty? should we check that?

**Comment by bdattoma** [comment]
could you also update the [test-variables.yml.example](https://github.com/red-hat-data-services/ods-ci/blob/master/ods_ci/test-variables.yml.example)?

**Comment by jgarciao** [comment]
> could you also update the [test-variables.yml.example](https://github.com/red-hat-data-services/ods-ci/blob/master/ods_ci/test-variables.yml.example)?

It is updated already https://github.com/red-hat-data-services/ods-ci/pull/873/files#diff-306ca20729ca75fa6e8aabe54d75fbb28f1c064a85c5674f8aa2e16834b5345f

**Review by bdattoma — APPROVED** [review]

File: ods_ci/tests/Resources/RHOSi.resource [hunk 3]
```diff
@@ -71,3 +72,11 @@ Initialize Global Variables
...
+    Variable Should Exist    ${APPLICATIONS_NAMESPACE}
+    Variable Should Exist    ${MONITORING_NAMESPACE}
+    Variable Should Exist    ${OPERATOR_NAMESPACE}
+    Variable Should Exist    ${NOTEBOOKS_NAMESPACE}
```
[the remark this answers is not in the document]
Comment (jgarciao) [reply ods_ci/tests/Resources/RHOSi.resource:82]: I just tested it. If the value is empty `Variable Should Exist` won't fail

**Comment by jgarciao** [comment]
> Due that we are going to support ODH and RHODS Maybe the function could be:
> 
> Before:
> 
> ```
> Is RHODS Version Greater Or Equal Than
> ```
> 
> After:
> 
> ```
> Is Version Greater Or Equal Than
> ```
> 
> It applies to the other functions that have RHODS in the name
I thought about this but I prefer to do it in a follow-up PR. I wanted to have this initial PR as little as possible in order to merge it faster

**Pull request merged**